src/reflect/scala/reflect/internal/util/Statistics.scala

```diff
@@ -250,13 +257,47 @@ quant)
   private var _enabled = false
   private val qs = new mutable.HashMap[String, Quantity]
 
-  /** replace with
-   *
-   *    final val canEnable = false
-   *
-   *  to remove all Statistics code from build
+  private var switchPoint: SwitchPoint = new SwitchPoint()
+  private final val DefaultHandle: MethodHandle = {
+    val handle = MethodHandles.lookup().findVirtual(this.getClass, "defaultValue", StatisticsHelper.BooleanMethodType)
+    val bounded = handle.bindTo(this)
+    setTarget(bounded)
+    bounded
+  }
```

Comment:
I'd suggest to put all the method handle contants into a Java source file so they can be try `static final` constants. 

https://wiki.openjdk.java.net/display/HotSpot/Deconstructing+MethodHandles

> If such a reference to a MethodHandle is held in a static final field then the runtime should be able to constant fold invocations on that reference and what it holds when inlining occurs.